src/plots/cartesian/set_convert.js

```diff
@@ -184,10 +184,13 @@ module.exports = function setConvert(ax) {
             // that aren't in the first etc.
             // TODO: sorting options - do the sorting
             // progressively here as we insert?
-            if(ax._categories.indexOf(v)===-1) ax._categories.push(v);
 
-            var c = ax._categories.indexOf(v);
-            return c===-1 ? constants.BADNUM : c;
+            if(v !== null && v !== undefined && ax._categories.indexOf(v) === -1){
```

Comment:
nice, maybe we should make that standard: http://eslint.org/docs/rules/no-eq-null